obspy/io/sac/util.py

```diff
@@ -0,0 +1,357 @@
+
+    References
+    ----------
+    http://docs.scipy.org/doc/numpy/user/basics.byteswapping.html
+
+    """
+    return [arr.newbyteorder('S') for arr in arrays]
+
+
+def is_same_byteorder(bo1, bo2):
+    """
+    Deal with all the ways to compare byte order string representations.
+
+    Parameters
+    ----------
+    bo1, bo2 : str
+        Byte order strings can be any of:
+        {'l', 'little', 'L', '<', 'b', 'big', 'B', '>', 'n', 'native', 'N', '='}
+
+    Returns
+    -------
+    isSAME : bool
+
+    """
+    # TODO: extend this as is_same_byteorder(*byteorders)
+    BE = ('b', 'big', '>')
+    LE = ('l', 'little', '<')
+    NE = ('n', 'native', '=')
+    OK = BE + LE + NE
+
+    if (bo1.lower() not in OK) or (bo2.lower() not in OK):
+        raise ValueError("Unrecognized byte order string.")
+
+    # make native decide what it is
+    bo1 = sys.byteorder if bo1.lower() in NE else bo1
+    bo2 = sys.byteorder if bo2.lower() in NE else bo2
+
+    return (bo1.lower() in LE) == (bo2.lower() in LE)
+
+
+def _clean_str(value):
+    null_term = value.find('\x00')
+    if null_term >= 0:
+        value = value[:null_term]
+    value = value.strip()
+
+    return value
+
+# TODO: do this in SACTrace?
+def sac_to_obspy_header(sacheader):
+    """
+    Make an ObsPy Stats header dictionary from a SAC header dictionary.
+
+    """
+    # 1. get required sac header values
+    try:
+        npts = sacheader['npts']
+        delta = sacheader['delta']
+    except KeyError:
+        msg = "Incomplete SAC header information to build an ObsPy header."
+        raise KeyError(msg)
+
+    assert npts != HD.INULL
+    assert delta != HD.FNULL
+    #
+    # 2. get time
+    try:
+        reftime = get_sac_reftime(sacheader)
+    except (SacError, ValueError):
+        #reftime = datetime.datetime(1970, 1, 1, 0, 0)
+        reftime = UTCDateTime(0.0)
+
+    b = sacheader.get('b', HD.FNULL)
+    #
+    # 3. get optional sac header values
+    calib = sacheader.get('scale', HD.FNULL)
+    kcmpnm = sacheader.get('kcmpnm', HD.SNULL)
+    kstnm = sacheader.get('kstnm', HD.SNULL)
+    knetwk = sacheader.get('knetwk', HD.SNULL)
+    khole = sacheader.get('khole', HD.SNULL)
+    #
+    # 4. deal with null values
+    b = b if (b != HD.FNULL) else 0.0
+    calib = calib if (calib != HD.FNULL) else 1.0
+    kcmpnm = kcmpnm if (kcmpnm != HD.SNULL) else ''
+    kstnm = kstnm if (kstnm != HD.SNULL) else ''
+    knetwk = knetwk if (knetwk != HD.SNULL) else ''
+    khole = khole if (khole != HD.SNULL) else ''
+    #
+    # 5. transform to obspy values
+    # nothing is null
+    stats = {}
+    stats['npts'] = npts
+    stats['sampling_rate'] = np.float32(1.) / np.float32(delta)
+    stats['network'] = _clean_str(knetwk)
+    stats['station'] = _clean_str(kstnm)
+    stats['channel'] = _clean_str(kcmpnm)
+    stats['location'] = _clean_str(khole)
+    stats['calib'] = calib
+
+    # store _all_ provided SAC header values
+    stats['sac'] = sacheader.copy()
```

Comment:
I did, but forgot to follow up- sorry.  Yes, unset headers return -12345 when `debug_headers=True`.  If `False`, they just don't show up.  There are some [tests](https://github.com/jkmacc-LANL/obspy/blob/master/obspy/io/sac/tests/test_core.py#L424) that ensure this, too.